testsuite/tests/lib-threads/uncaught_exception_handler.ml

```diff
@@ -0,0 +1,34 @@
+(* TEST
+
+flags = "-g"
+ocamlrunparam += ",b=1"
+
+* hassysthreads
+include systhreads
+** bytecode
+** native
+
+*)
+
+(* Testing if uncaught exception handlers are behaving properly  *)
+
+let () = Printexc.record_backtrace true
+
+let handler exn =
+  let id = Thread.self () |> Thread.id in
+  let msg = Printexc.to_string exn in
+  Printf.eprintf "[thread %d] caught %s\n" id msg;
+  Printexc.print_backtrace stderr;
+  flush stderr;
+  raise exn
```

Comment:
The test result is slightly harder to make sense of when it throws the same exception!